datafusion/core/tests/fuzz_cases/aggregation_fuzzer/context_generator.rs

```diff
@@ -0,0 +1,342 @@
+    sort_hint: bool,
+    skip_partial_params: SkipPartialParams,
+    table_name: String,
+    table_provider: Arc<dyn TableProvider>,
+}
+
+impl GeneratedSessionContextBuilder {
+    fn build(self) -> Result<SessionContextWithParams> {
+        // Build session context
+        let mut session_config = SessionConfig::default();
+        session_config = session_config.set(
+            "datafusion.execution.batch_size",
+            &ScalarValue::UInt64(Some(self.batch_size as u64)),
+        );
+        session_config = session_config.set(
+            "datafusion.execution.target_partitions",
+            &ScalarValue::UInt64(Some(self.target_partitions as u64)),
+        );
+        session_config = session_config.set(
+            "datafusion.execution.skip_partial_aggregation_probe_rows_threshold",
+            &ScalarValue::UInt64(Some(self.skip_partial_params.rows_threshold as u64)),
+        );
+        session_config = session_config.set(
+            "datafusion.execution.skip_partial_aggregation_probe_ratio_threshold",
+            &ScalarValue::Float64(Some(self.skip_partial_params.ratio_threshold)),
+        );
+
+        let ctx = SessionContext::new_with_config(session_config);
+        ctx.register_table(self.table_name, self.table_provider)?;
+
+        let params = SessionContextParams {
+            batch_size: self.batch_size,
+            target_partitions: self.target_partitions,
+            sort_hint: self.sort_hint,
+            skip_partial_params: self.skip_partial_params,
+        };
+
+        Ok(SessionContextWithParams { ctx, params })
+    }
+}
+
+/// The generated params for [`SessionContext`]
+#[derive(Debug)]
+#[allow(dead_code)]
+pub struct SessionContextParams {
+    batch_size: usize,
+    target_partitions: usize,
+    sort_hint: bool,
+    skip_partial_params: SkipPartialParams,
+}
+
+/// Partial skipping parameters
+#[derive(Debug, Clone, Copy)]
+pub struct SkipPartialParams {
+    /// Related to `skip_partial_aggregation_probe_ratio_threshold` in `ExecutionOptions`
+    pub ratio_threshold: f64,
+
+    /// Related to `skip_partial_aggregation_probe_rows_threshold` in `ExecutionOptions`
+    pub rows_threshold: usize,
+}
+
+impl SkipPartialParams {
+    /// Generate `SkipPartialParams` ensuring to trigger partial skipping
+    pub fn ensure_trigger() -> Self {
+        Self {
+            ratio_threshold: 0.0,
+            rows_threshold: 0,
+        }
+    }
+
+    /// Generate `SkipPartialParams` ensuring not to trigger partial skipping
+    pub fn ensure_not_trigger() -> Self {
+        Self {
+            ratio_threshold: 1.0,
+            rows_threshold: usize::MAX,
+        }
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use arrow_array::{RecordBatch, StringArray, UInt32Array};
+    use arrow_schema::{DataType, Field, Schema};
+
+    use crate::fuzz_cases::aggregation_fuzzer::check_equality_of_batches;
+
+    use super::*;
+
+    #[tokio::test]
+    async fn test_generated_context() {
```

Comment:
I wonder if this is still needed now that you have the other fuzz test like test_group_by_mixed_string_int64 🤔

Reply:
> I wonder if this is still needed now that you have the other fuzz test like test_group_by_mixed_string_int64 🤔

It is the unit test with a fixed dataset for decoupling from `DatasetGenerator`, so we can check the correctness of `SessionContextGenerator`, maybe we still need it?